crates/evm/evm.rs

```diff
@@ -59,11 +69,56 @@ fn run_evm(
 }
 
 // Merges transitions stored when executing transactions and applies the resulting changes to the DB
-pub fn apply_state_transitions(state: &mut EvmState) {
-    state.0.merge_transitions(BundleRetention::Reverts);
-    let _bundle = state.0.take_bundle();
-    // TODO: Apply bundle to DB
-    unimplemented!("Apply state transitions to DB")
+pub fn apply_state_transitions(state: &mut EvmState) -> Result<(), StoreError> {
+    state.0.merge_transitions(BundleRetention::PlainState);
+    let bundle = state.0.take_bundle();
+    // Update accounts
+    for (address, account) in bundle.state() {
+        if account.status.is_not_modified() {
+            continue;
+        }
+        let address = Address::from_slice(address.0.as_slice());
+        // Remove account from DB if destroyed
+        if account.status.was_destroyed() {
+            state.database().remove_account(address)?;
+        }
+        // Apply account changes to DB
+        // If the account was changed then both original and current info will be present
```

Comment:
I don't understand this. If the old account and the new one have the same key (address), won't the new one replace the older one?

Reply:
The comment refers to the `BundleAccount` struct from revm that contains optional fields for the current and original info of the account. This refers to as why we handle it with `if let Some` in the code but don't handle the case of `None`. Ill clarify the comment to make it clear that it refers to the bundle account